src/tests/tracing/eventpipe/common/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,195 @@
+                case TransportType.NamedPipe:
+                {
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{NamedPipeSchema} is only supported on Windows.");
+                    break;
+                }
+                case TransportType.UnixDomainSocket:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{UnixDomainSocketSchema} is not supported on Windows, use {NamedPipeSchema}.");
+                    break;
+                }
+#if DIAGNOSTICS_RUNTIME
+                case TransportType.TcpSocket:
+                {
+                    break;
+                }
+#endif
+                default:
+                {
+                    throw new NotSupportedException($"{transportType} not supported.");
+                }
+            }
+
+            Address = address;
+            _transportType = transportType;
+            _portType = portType;
+        }
+
+        // Config format: [Address],[PortType]
+        //
+        // Address in UnixDomainSocket formats:
+        // myport => myport
+        // uds:myport => myport
+        // /User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds:/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds://authority/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds:///User/mrx/myport.sock => /User/mrx/myport.sock
+        //
+        // Address in NamedPipe formats:
+        // myport => myport
+        // namedpipe:myport => myport
+        // \\.\pipe\myport => myport (dropping \\.\pipe\ is inline with implemented namedpipe client/server)
+        // namedpipe://./pipe/myport => myport (dropping authority and /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe:/pipe/myport  => myport (dropping /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe://authority/myport => myport
+        // namedpipe:///myport => myport
+        //
+        // PortType: Listen|Connect, default Listen.
+        public static bool TryParse(string config, out IpcEndpointConfig result)
+        {
+            try
+            {
+                result = Parse(config);
+            }
+            catch(Exception)
+            {
+                result = null;
+            }
+            return result != null;
+        }
+
+        public static IpcEndpointConfig Parse(string config)
+        {
+            if (string.IsNullOrEmpty(config))
+                throw new FormatException("Missing IPC endpoint config.");
+
+            string address = "";
+            PortType portType = PortType.Connect;
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : TransportType.UnixDomainSocket;
+#if DIAGNOSTICS_RUNTIME
+            if (OperatingSystem.IsAndroid())
```

Comment:
Is this only true if the lib is running directly on android? Where is the tracer app running?

Reply:
This is true for our runtime tests when building with a TargetOS of Android (in Dotnet/Diagnostics, `OperatingSystem.IsAndroid()` isn't available as the repo depends on an older version of the runtime i believe)
I *think* whats happening is that we enable `dotnet-dsrouter` to connect the runtime with diagnostic tools through a TCP/IP connection. Then tools such as `dotnet-counters` and `dotnet-trace` can be ran on the local machine to get events from the device/simulator.

Reply:
Maybe it is better to not do this check like this, instead look at the address and if we are using UnixDomainSocket but the address looks like a TCP/IP address, set transport type to TransportType.TcpSocket, that would eliminate the need to look for Android, since TCP/IP can be supported on more platforms going forward (at least iOS).

Reply:
[Earlier in the comments](https://github.com/dotnet/runtime/pull/64358#issuecomment-1029647629) I proposed that we wouldn't make any changes to Parse(), and thus there would be no string (for now) that parses to a tcp endpoint. My motivation was that if we aren't ready to expose TCP as an end-user feature yet and this is only for testing then there is no need to design a parsing mechanism in advance. Of course if you disagree by all means we can discuss, but I worried we might just be talking past each other and creating some confusion : )

Reply:
OK, now I saw that comment, missed it. So your idea was to move all out to the client and leave library as is:

`DiagnosticsClient(new DiagnosticPortIpcEndpoint(new IpcEndpointConfig("127.0.0.1:9000", TransportType.TcpSocket, PortType.Listen)));`

and put logic on what transport to use on client side not include any additional support except getting above statement to work through diagnostic client library.

I thought getting the logic into IpcEndPoint was low hanging fruit (since we already put effort into analyze the change) inline with how it would be done going forward and guard those code paths with the DIAGNOSTICS_RUNTIME so it won't be usable from diagnostic repro, but I agree that we could push the logic out of the library and into the test client.

Reply:
I went forward with not modifying `Parse()` and instead directly utilizing the internal `DiagnosticsClient(IpcEndpointConfig config)` constructor instead of the `DiagnosticsClient(IpcEndpoint endpoint)` (since it is more convenient, though not as direct, but if there are any drawbacks I can utilize the endpoint constructor instead).
I also removed the string constructor and added a comment above `IpcEndpointConfig.Address`